module/zfs/zvol.c

```diff
@@ -1361,9 +1334,7 @@ zvol_open(struct block_device *bdev, fmode_t flag)
 	int error = 0;
 	boolean_t drop_suspend = B_TRUE;
 
-	ASSERT(!MUTEX_HELD(&zvol_state_lock));
-
-	mutex_enter(&zvol_state_lock);
+	rw_enter(&zvol_state_lock, RW_READER);
```

Comment:
Can we assert that zvol_state_lock is not held here ?

Reply:
Since this was converted to a krwlock it's now possible that another thread is holding the lock as a reader.  The best we can do is assert that we're not holding it a writer but that wasn't really useful so I dropped it entirely.

Reply:
Right, it makes sense, thanks.